tests/test-cast-to-underflow.c

```diff
@@ -34,6 +34,14 @@
 #define _WANT_VC 1
 #include "scaffold.c"
 
+/* TODO: libdfp conversion routines are not correct to 1ulp.
+         It would seem s390 hardware routines are more accurate. */
+#if !defined(__s390x)
+#define XFAIL_CONVERSION 1
+#else
+#define XFAIL_CONVERSION 0
+#endif
+
```

Comment:
XFAIL_CONVERSION should be "0" if we run on s390/s390x and if hardware dfp is available but not if we use software dfp.
In either case, the macro "__s390x" is never defined, but instead the following macros are defined on:
-s390 (31bit): __s390__
-s390x (64bit): __s390__ and __s390x__
Thus you can detect s390 (31 or 64bit) if __s390__ is defined.
The information if hardware dfp is used is currently not available via a macro definition from gcc or e.g. in config.h.
But we know it in configure.ac as with_dfp is set to yes or no.